File: Java/1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hashmap = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++){
            if(hashmap.containsKey(nums[i])){
                int t = hashmap.get(nums[i]);
                return new int[]{t, i};
            }
            else{
                hashmap.put(target - nums[i], i);
            }
        }
        return new int[2];
    }
}

File: Rust/1. Two Sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut temp: HashMap<i32, usize> = HashMap::new();

        for (i, j) in nums.iter().enumerate() {
            if temp.contains_key(j) {
                return vec![temp.get(j).unwrap().clone() as i32, i as i32];
            }
            else {
                temp.insert(target - j, i);
            }
        }

        return vec![0, 0];
    }
}
